subprojects/packagefiles/openblas/meson.build

```diff
@@ -597,53 +630,62 @@ template = f'@_msproot@/openblas_config_template.h'
 common_interface = f'@_msproot@/common_interface.h'
 cblas = f'@_msproot@/cblas.h'
 
-# Prepare config_last.h from config.h
-pcl = custom_target('prepare_config_last',
-  output : 'config_last.h',
-  command : [py3, f'@_msproot@/prepare_config_last.py',
-             '--config', fconfig_h,
-             '--output', fconfig_last,
-             quad_prec ? '--quad-precision' : [],
-             exprecision ? '--exprecision' : [],
-            ],
-  build_by_default : true,
-)
-
-# Generate the headers
-custom_target('gen_install_headers',
-  output : ['openblas_config.h', 'f77blas.h', 'cblas.h'],
-  command : [py3, f'@_msproot@/gen_install_headers.py',
-             '--dest-dir', f'@_mbproot@',
-             '--version', meson.project_version(),
-             '--config-last', fconfig_last,
-             '--template', template,
-             '--common-interface', common_interface,
-             '--cblas', cblas,
-             not get_option('no_fortran') ? '--generate-f77blas' : [],
-             not get_option('build_without_cblas') ? '--generate-cblas' : [],
-            ],
-  install : true,
-  install_dir : 'include',
-  depends : pcl,
-)
-
-pkg.generate(openblas,
-             name: 'openblas',
-             filebase: 'meson-openblas',
-             description: 'OpenBLAS via meson build',
-             version: f'@pkg_ver@_meson',
-             install_dir: pkg_install_dir,
-            )
+if not as_subproject
+  # Prepare config_last.h from config.h
+  pcl = custom_target('prepare_config_last',
+    output : 'config_last.h',
+    command : [py3, f'@_msproot@/prepare_config_last.py',
+               '--config', fconfig_h,
+               '--output', fconfig_last,
+               quad_prec ? '--quad-precision' : [],
+               exprecision ? '--exprecision' : [],
+              ],
+    build_by_default : true,
+  )
+
+  # Generate the headers
+  custom_target('gen_install_headers',
+    output : ['openblas_config.h', 'f77blas.h', 'cblas.h'],
+    command : [py3, f'@_msproot@/gen_install_headers.py',
+               '--dest-dir', f'@_mbproot@',
+               '--version', meson.project_version(),
+               '--config-last', fconfig_last,
+               '--template', template,
+               '--common-interface', common_interface,
+               '--cblas', cblas,
+               not get_option('no_fortran') ? '--generate-f77blas' : [],
+               not get_option('build_without_cblas') ? '--generate-cblas' : [],
+              ],
+    install : true,
+    install_dir : 'include',
+    depends : pcl,
+  )
+
+  pkg.generate(openblas,
+               name: 'openblas',
+               filebase: 'meson-openblas',
+               description: 'OpenBLAS via meson build',
+               version: f'@pkg_ver@_meson',
+               install_dir: pkg_install_dir,
+              )
+endif
 
-if build_testing
+if build_testing and not as_subproject
```

Comment:
This is already disabled by default. Why prevent people from enabling tests when building as a subproject? Maybe they enabled both options because they want to test the dependencies compiled correctly too? :)

Reply:
Ah, not sure why I had to disable this, `build_testing` should default to `false` indeed.

Reply:
Addressed now, resolving.